### PR TITLE
Ghost7/twitch client

### PR DESCRIPTION
**Pull request by ghost7**

### CODE REVIEW

**Comment by marknach** [comment]
:O

File: TwitchClient/Client/BaseClient.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
+﻿using CoffeeCat.RiotCommon.Utils;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace TwitchClient.Client
```
Comment (tmajest) [inline TwitchClient/Client/BaseClient.cs]: Could you update the assembly name and default namespace to use the CoffeeCat prefix?

:D

File: TwitchClient/Endpoints/Channels/ChannelVideosEndpoint.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        public ChannelVideosEndpoint(string channelName) : base(channelName)
+        {
+        }
+
+        public override string ChannelResource => "videos";
+
+        protected override IDictionary<string, string> CreateQueryParameters()
+        {
+            return new Dictionary<string, string>()
```
Comment (tmajest) [inline TwitchClient/Endpoints/Channels/ChannelVideosEndpoint.cs:35]: sexy

File: TwitchClient/Endpoints/TwitchEndpoint.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        public abstract string Format();
+
+        protected abstract IDictionary<string, string> CreateQueryParameters();
+
+        public Uri GetUri()
+        {
+            var apiBaseFormatted = this.Format();
+
+            var queryParameterDict = this.CreateQueryParameters();
+            var joined = queryParameterDict.Where(kv => kv.Value != null).Select(kv => string.Format("{0}={1}", kv.Key, kv.Value));
```
Comment (tmajest) [inline TwitchClient/Endpoints/TwitchEndpoint.cs]: I've always wanted an IEnumerable extension method that filters out null items.  Could you write this?  Maybe have a second one that takes in a selector function too for your case above.

**Review by tmajest — APPROVED** [review]

File: TwitchClient/Endpoints/TwitchEndpoint.cs (hunk 3 above)
Comment (ghost7) [reply TwitchClient/Endpoints/TwitchEndpoint.cs]: In this case, I'm checking if the value property is not null, not the actual object. Do you want an extension for that?

Comment (tmajest) [reply TwitchClient/Endpoints/TwitchEndpoint.cs]: The selector function could select the thing that you're checking for null

**Pull request merged**